homeassistant/components/automation/__init__.py

```diff
@@ -395,10 +395,8 @@ def referenced_entities(self):
 
         try:
             await self.action_script.async_run(variables, trigger_context)
-        except Exception as err:  # pylint: disable=broad-except
-            self.action_script.async_log_exception(
-                _LOGGER, f"Error while executing automation {self.entity_id}", err
-            )
+        except Exception:  # pylint: disable=broad-except
+            pass
```

Comment:
We shouldn't swallow all exceptions. We should catch the ones we expect and handle correctly (like ignoring with `pass`). We should not go and catch everything, as this will swallow attribute errors, type errors etc that are real bugs.

Reply:
This is effectively the way it was, except that the logging is now done entirely within the script helper, as opposed to requiring the "user" (in this case, the automation integration) having to call back into the script helper to get the error logged.

To put that another way, the automation integration was already catching all exceptions and just logging them. I did not change that behavior.

Now, having said that, if you still want to change the behavior and have the automation integration not catch anything, it doesn't really matter to me either way, and I can certainly do that.

Reply:
The reason we want logging to be called from Automation/Script is because that means we give the caller control about logging/error handling, and the caller can also add a reference to who owns the script (Automation X got an exception, Script Y got an exception)

Reply:
The idea you had suggested before was to create a logger specific to the automation or script, e.g., `logging.getLogger(f"{__name__}.{self.name}")` and pass that into the script helper to use. That would satisfy that need. (E.g., the logger might be `homeassistant.components.automation.hello` instead of just `homeassistant.components.automation`.) I was trying to keep the changes in this PR limited to the script helper, while only changing as little elsewhere as possible. That's why I haven't yet fully implemented your earlier suggestion.

In the end, the script helper takes care of logging the exception (on the caller's behalf, by using the logger it passed in) rather than requiring the caller to make a callback into the script helper to log the error appropriately. That is much cleaner and much less error prone (because now there is no longer the issue of the saved exception state, like what step it was on, etc., getting lost.) And since the exception is re-raised, the caller still has the option to decide what to do with it (other than logging, which has been done for it.)